### PR TITLE
Add Flax example tests

### PR DESCRIPTION
**Pull request by patil-suraj**
# What does this PR do?

This PR adds tests for flax examples. 

The test for image-classification example is not added since the script uses a manual dataset with `torchvision`.  The test will be added once the script is refactored to use `datasets`.

### CODE REVIEW

File: examples/flax/test_examples.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
+# coding=utf-8
```
Comment (patrickvonplaten) [inline examples/flax/test_examples.py:1]: Great!

File: examples/flax/language-modeling/run_t5_mlm_flax.py [hunk 2]
```diff
@@ -522,9 +523,7 @@ def write_eval_metric(summary_writer, eval_metrics, step):
             model_args.config_name, cache_dir=model_args.cache_dir, vocab_size=len(tokenizer)
         )
     elif model_args.model_name_or_path:
-        config = T5Config.from_pretrained(
-            model_args.model_name_or_path, cache_dir=model_args.cache_dir, vocab_size=len(tokenizer)
-        )
+        config = T5Config.from_pretrained(model_args.model_name_or_path, cache_dir=model_args.cache_dir)
```
Comment (patil-suraj) [inline examples/flax/language-modeling/run_t5_mlm_flax.py:526]: When loading `config` from pre-trained models, we shouldn't set the `vocab_size` here.
This results in an error as there is a mismatch in `len(tokenizer)` and the `vocab_size` used in the pre-trained T5 models. Which is why right now it's not possible to load a pre-trained model with this script.

Comment (patrickvonplaten) [reply examples/flax/language-modeling/run_t5_mlm_flax.py:526]: I am a bit worried though that people will do the same mistake that was made for GPT2 pretraining during the Flax/JAX sprint when a model is trained from scratch with a tokenizer that is larger than the model. In such a case it is possible that training can go for a long time until it errors out. 

On the other hand, I definitely see the need to remove this!

Could we maybe do the following:

remove `len(tokenizer)` here just as you did but then add the following:

```
config.vocab_size = len(tokenizer)
```

under 

```=
model = FlaxT5ForConditionalGeneration(config, seed=training_args.seed, dtype=getattr(jnp, model_args.dtype))
```

which is only called for "from scratch" pretraining? This way we are save for both cases.

**Review by patrickvonplaten — APPROVED** [review]
Nice - good job! Would be nice to make T5 pretraining a bit more robust as described in the comment above.

**Pull request merged**